core/http-auth-aws-crt/src/main/java/software/amazon/awssdk/http/auth/aws/crt/AwsCrtV4aHttpSigner.java

```diff
@@ -27,6 +30,73 @@
 @SdkPublicApi
 public interface AwsCrtV4aHttpSigner extends HttpSigner<AwsCredentialsIdentity> {
 
+    /**
+     * The AWS region name to be used for computing the signature.
+     * This property is required.
+     */
+    SignerProperty<String> REGION_NAME =
+        SignerProperty.create(String.class, "RegionName");
+
+    /**
+     * The name of the AWS service.
+     * This property is required.
+     */
+    SignerProperty<String> SERVICE_SIGNING_NAME =
+        SignerProperty.create(String.class, "ServiceSigningName");
+
+    /**
+     * A {@link Clock} to be used at the time of signing.
+     * This property defaults to the time at which signing occurs.
+     */
+    SignerProperty<Clock> SIGNING_CLOCK =
+        SignerProperty.create(Clock.class, "SigningClock");
+
+    /**
+     * The name of the header for the checksum.
+     * This property is optional.
+     */
+    SignerProperty<String> CHECKSUM_HEADER_NAME =
+        SignerProperty.create(String.class, "ChecksumHeaderName");
```

Comment:
When would someone specify this?

Reply:
When a checksum is already added to the request before signing, or when they want the signer to use a different checksum algorithm (such as crc32)